include/exadg/solvers_and_preconditioners/nonlinear_solvers/newton_solver.h

```diff
@@ -114,22 +145,24 @@ class Solver
       } while(norm_r_damp >= (1.0 - tau * omega) * norm_r && n_iter_damp < max_iter_damp);
 
       AssertThrow(norm_r_damp < (1.0 - tau * omega) * norm_r,
-                  dealii::ExcMessage("Damped Newton iteration did not converge. "
-                                     "Maximum number of iterations exceeded!"));
+                  dealii::ExcMessage("Damped nonlinear iteration did not converge. "
+                                     "Maximum number of iterations (" +
+                                     std::to_string(max_iter_damp) + ") exceeded!"));
 
       // update solution and residual
-      solution = temporary;
+      solution = temporary_solution;
       norm_r   = norm_r_damp;
 
       // increment iteration counter
-      ++newton_iterations;
+      ++nonlinear_iterations;
       linear_iterations += n_iter_linear;
     }
 
     AssertThrow(norm_r <= this->solver_data.abs_tol || norm_r / norm_r_0 <= solver_data.rel_tol,
                 dealii::ExcMessage(
-                  "Newton solver failed to solve nonlinear problem to given tolerance. "
-                  "Maximum number of iterations exceeded!"));
+                  "Nonlinear solver failed to solve nonlinear problem to given tolerance. "
+                  "Maximum number of iterations (" +
+                  std::to_string(solver_data.max_iter) + ") exceeded!"));
```

Comment:
If we want to use linearized solvers, we have to give the nonlinear solvers additional parameters: 'enable_damping' (and more parameters potentially, these are hard-coded above), and some switch to activate the Assertion here, such that doing 5 nonlinear iters is ok, no matter the reduction in the residual.